src/validation/statement.rs

```diff
@@ -151,6 +151,29 @@ pub fn visit_statement<T: AnnotationMap>(
                         .with_location(statement.get_location()).with_error_code("E119"));
             }
         }
+        AstStatement::This => {
+            if !context.qualifier.is_some_and(|it| {
+                context
+                    .index
+                    .find_pou(it)
+                    .and_then(|it| {
+                        if it.is_function_block() {
+                            Some(it)
+                        } else {
+                            context.index.find_pou(it.get_parent_pou_name().unwrap_or_default())
```

Comment:
I think this is wrong? You only want to return the POU in the `else` block if we're dealing with a method?
Something along perhaps
```
.and_then(|it| match it {
    PouIndexEntry::FunctionBlock { .. } => Some(it),
    PouIndexEntry::Method { parent_name, .. } => context.index.find_pou(&parent_name),
    _ => None,
})
```
(Ignore if `THIS` access in ACTIONs is actually supported according to the norm / codesys though)